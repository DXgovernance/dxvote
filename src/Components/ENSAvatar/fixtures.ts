import { ENSAvatarProps } from './types';

export const withAvatarNFT: ENSAvatarProps = {
  address: '0x0b17cf48420400e1D71F8231d4a8e43B3566BB5B',
};

export const withoutAvatarNFT: ENSAvatarProps = {
  address: '0x519b70055af55A007110B4Ff99b0eA33071c720a',
};

export const customSize: ENSAvatarProps = {
  ...withAvatarNFT,
  size: 100,
};
