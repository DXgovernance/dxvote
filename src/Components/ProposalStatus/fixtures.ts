import { ProposalState } from 'Components/Types';
import moment from 'moment';
import { ProposalStatusProps } from './types';

export const proposalStatusMock: ProposalStatusProps = {
  endTime: moment(),
  status: ProposalState.Active,
  timeDetail: '1 minute ago',
};
