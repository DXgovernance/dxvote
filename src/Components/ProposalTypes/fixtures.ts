import { ProposalTypesProps } from './types';

export const testProps: ProposalTypesProps = {
  data: [
    {
      title: 'Signal Proposal',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed leo quam, blandit eu sapien eu, commodo dapibus nisl',
      onChainAction: false,
    },
    {
      title: 'Custom Proposal',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed leo quam, blandit eu sapien eu, commodo dapibus nisl',
      onChainAction: false,
    },
  ],
};
