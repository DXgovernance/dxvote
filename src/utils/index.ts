export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
