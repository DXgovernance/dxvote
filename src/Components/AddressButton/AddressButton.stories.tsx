import { ComponentStory, ComponentMeta } from '@storybook/react';
import AddressButton from 'Components/AddressButton/AddressButton';
import { fullProps, partialProps } from './fixtures';

export default {
  title: 'Addresses/AddressButton',
  component: AddressButton,
  argTypes: {
    address: {
      description: 'Ethereum address',
    },
    transactionsCounter: {
      description: 'Number of transactions for the address',
    },
  },
} as ComponentMeta<typeof AddressButton>;

const Template: ComponentStory<typeof AddressButton> = args => (
  <AddressButton {...args} />
);

export const WithTransactions = Template.bind({});
WithTransactions.args = fullProps;

export const WithoutTransactions = Template.bind({});
WithoutTransactions.args = partialProps;

export const Loading = Template.bind({});
Loading.args = {};
