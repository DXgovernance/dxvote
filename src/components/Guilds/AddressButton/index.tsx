import { isDesktop } from 'react-device-detect';
import styled from 'styled-components';
import React, { useMemo } from 'react';
import Skeleton from 'react-loading-skeleton';
import { IconButton } from '../common/Button';
import useENSAvatar from '../../../hooks/Guilds/ens/useENSAvatar';
import Avatar from '../Avatar';
import { shortenAddress } from '../../../utils';
import { Badge } from '../common/Badge';
import { DEFAULT_ETH_CHAIN_ID } from 'provider/connectors';

const IconHolder = styled.span`
  display: flex;
  justify-content: center;

  @media only screen and (min-width: 768px) {
    margin-right: 0.3rem;
  }

  img {
    border-radius: 50%;
    margin-right: 0;
  }
`;

const StyledAddressButton = styled(IconButton)`
  margin-top: 0;
  margin-bottom: 0;
  padding: 0.3rem;

  @media only screen and (min-width: 768px) {
    padding: 0.3rem 0.5rem;
  }

  /* hover state for when having child Badge */
  &:hover,
  &:active {
    ${Badge} {
      background-color: ${({ theme }) => theme.colors.background};
      color: ${({ theme }) => theme.colors.primary};
    }
  }
`;

const AddressText = styled.span`
  margin-left: 0.2rem;
  margin-right: 0.3rem;
`;

interface AddressButtonProps {
  address?: string;
  chainId?: number;
  transactionsCounter?: number;
  onClick?: () => void;
}

const AddressButton: React.FC<AddressButtonProps> = ({
  address,
  chainId = DEFAULT_ETH_CHAIN_ID,
  transactionsCounter,
  onClick,
}) => {
  const { ensName, imageUrl, avatarUri } = useENSAvatar(address, chainId);

  const imageUrlToUse = useMemo(() => {
    if (avatarUri) {
      // TODO: Consider chainId when generating ENS metadata service fallback URL
      return (
        imageUrl || `https://metadata.ens.domains/mainnet/avatar/${ensName}`
      );
    } else {
      return null;
    }
  }, [imageUrl, ensName, avatarUri]);

  return (
    <StyledAddressButton onClick={onClick} iconLeft>
      <IconHolder>
        {address ? (
          <Avatar src={imageUrlToUse} defaultSeed={address} size={24} />
        ) : (
          <Skeleton circle width={24} height={24} />
        )}
      </IconHolder>
      {isDesktop && (
        <AddressText>
          {ensName || address ? (
            shortenAddress(address)
          ) : (
            <Skeleton width={100} />
          )}
        </AddressText>
      )}
      {transactionsCounter && <Badge size="25">{transactionsCounter}</Badge>}
    </StyledAddressButton>
  );
};

export default AddressButton;
