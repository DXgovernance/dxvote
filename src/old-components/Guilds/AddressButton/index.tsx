import { isDesktop } from 'react-device-detect';
import styled from 'styled-components';
import React from 'react';
import { IconButton } from '../common/Button';
import useENSAvatar from '../../../hooks/Guilds/ether-swr/ens/useENSAvatar';
import Avatar from '../../../Components/Primitives/Avatar/Avatar';
import { shortenAddress } from '../../../utils';
import { Badge } from '../common/Badge';
import { Loading } from '../../../Components/Primitives/Loading';

const IconHolder = styled.span`
  display: flex;
  justify-content: center;

  @media only screen and (min-width: 768px) {
    margin-right: 0.3rem;
  }

  img {
    border-radius: 50%;
    margin-right: 0;
  }
`;

const StyledAddressButton = styled(IconButton)`
  margin-top: 0;
  margin-bottom: 0;
  padding: 0.3rem;

  @media only screen and (min-width: 768px) {
    padding: 0.3rem 0.5rem;
  }

  /* hover state for when having child Badge */
  &:hover,
  &:active {
    ${Badge} {
      background-color: ${({ theme }) => theme.colors.background};
      color: ${({ theme }) => theme.colors.primary};
    }
  }
`;

const AddressText = styled.span`
  margin-left: 0.2rem;
  margin-right: 0.3rem;
`;

interface AddressButtonProps {
  address?: string;
  transactionsCounter?: number;
  onClick?: () => void;
}

const AddressButton: React.FC<AddressButtonProps> = ({
  address,
  transactionsCounter,
  onClick,
}) => {
  const { ensName, imageUrl } = useENSAvatar(address, 1);

  return (
    <StyledAddressButton variant="secondary" onClick={onClick} iconLeft>
      <IconHolder>
        {address ? (
          <Avatar src={imageUrl} defaultSeed={address} size={24} />
        ) : (
          <Loading
            loading
            text
            skeletonProps={{ circle: true, width: '24px', height: '24px' }}
          />
        )}
      </IconHolder>
      {isDesktop && (
        <AddressText>
          {ensName || address ? (
            shortenAddress(address)
          ) : (
            <Loading loading text />
          )}
        </AddressText>
      )}
      {transactionsCounter ? (
        <Badge size={25}>{transactionsCounter}</Badge>
      ) : null}
    </StyledAddressButton>
  );
};

export default AddressButton;
