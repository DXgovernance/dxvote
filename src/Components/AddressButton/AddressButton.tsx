import { isDesktop } from 'react-device-detect';
import React from 'react';
import { shortenAddress } from 'utils';
import { Badge } from 'old-components/Guilds/common/Badge';
import { Loading } from 'Components/Primitives/Loading';
import { AddressButtonProps } from 'Components/AddressButton/types';
import {
  IconHolder,
  StyledAddressButton,
  AddressText,
} from 'Components/AddressButton/AddressButton.styled';
import ENSAvatar from 'Components/ENSAvatar/ENSAvatar';
import useENS from 'hooks/Guilds/ether-swr/ens/useENS';

const AddressButton: React.FC<AddressButtonProps> = ({
  address,
  transactionsCounter,
  onClick,
}) => {
  const { name: ensName } = useENS(address, 1);

  return (
    <StyledAddressButton variant="secondary" onClick={onClick} iconLeft>
      <IconHolder>
        <ENSAvatar address={address} />
      </IconHolder>
      {isDesktop && (
        <AddressText>
          {address ? (
            ensName || shortenAddress(address)
          ) : (
            <Loading loading text />
          )}
        </AddressText>
      )}
      {transactionsCounter ? (
        <Badge size={25}>{transactionsCounter}</Badge>
      ) : null}
    </StyledAddressButton>
  );
};

export default AddressButton;
