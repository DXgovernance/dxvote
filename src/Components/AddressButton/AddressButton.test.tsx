import AddressButton from './AddressButton';
import { render } from '../../utils/tests';
import { fullProps, partialProps } from './fixtures';

jest.mock('hooks/Guilds/ether-swr/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

jest.mock('hooks/Guilds/ether-swr/ens/useENS', () => ({
  __esModule: true,
  default: () => ({
    address: 'test',
    name: 'test.eth',
  }),
}));

fullProps.onClick = jest.fn();

describe('AddressButton', () => {
  it('Should render properly with full props', async () => {
    const { container } = render(<AddressButton {...fullProps} />);
    expect(container).toMatchSnapshot();
  });
  it('Should render properly with partial props', async () => {
    const { container } = render(<AddressButton {...partialProps} />);
    expect(container).toMatchSnapshot();
  });
});
