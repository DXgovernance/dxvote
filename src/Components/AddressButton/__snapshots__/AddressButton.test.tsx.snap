// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressButton Should render properly with full props 1`] = `
<div>
   
  <button
    class="sc-hKgJUU sc-eCstlR sc-iBPTik yNXFo fJzGbV KMmoQ"
  >
    <span
      class="sc-gKseQn fTGEco"
    >
      <span
        class="sc-jrAFXE fkvRJA"
      >
        <img
          class="sc-jSgvzq sc-pGacB iUxzmp ghyxMf"
          data-testid="avatar"
          size="24"
          src="test"
        />
      </span>
    </span>
    <span
      class="sc-fubCzh ePKPqF"
    >
      test.eth
    </span>
    <div
      class="sc-bdfBQB bfxPFN"
      size="25"
    >
      5
    </div>
  </button>
   
</div>
`;

exports[`AddressButton Should render properly with partial props 1`] = `
<div>
   
  <button
    class="sc-hKgJUU sc-eCstlR sc-iBPTik yNXFo fJzGbV KMmoQ"
  >
    <span
      class="sc-gKseQn fTGEco"
    >
      <span
        class="sc-jrAFXE fkvRJA"
      >
        <img
          class="sc-jSgvzq sc-pGacB iUxzmp ghyxMf"
          data-testid="avatar"
          size="24"
          src="test"
        />
      </span>
    </span>
    <span
      class="sc-fubCzh ePKPqF"
    >
      test.eth
    </span>
  </button>
   
</div>
`;
